### PR TITLE
Store locale on intake and confirm Spanish notifications send correctly

### PR DESCRIPTION
**Pull request by rickreyhsig**
https://www.pivotaltracker.com/story/show/187063473

### **Welcome Email**
Screen Shot 2024-02-19 at 4 27 16 PM

### **Accepted/Refund**
Screen Shot 2024-02-19 at 6 02 35 PM

### **Accepted/Owed**
Screen Shot 2024-02-19 at 6 08 24 PM

### **Rejected**
Screen Shot 2024-02-19 at 6 29 35 PM

How to test from console:
in `StateFile::AfterTransitionMessagingService`
```
Trigger email from console:
@intake = StateFileAzIntake.last
  or
@intake = StateFileNyIntake.last
@submission = @intake.efile_submissions.last
return_status_link = app.url_for(host: MultiTenantService.new(:statefile).host, controller: "state_file/questions/return_status", action: "edit", us_state: 'ny')

message = StateFile::AutomatedMessage::AcceptedRefund
body_args = { return_status_link: return_status_link }
      StateFile::MessagingService.new(
        intake: @intake,
        submission: @submission,
        message: message,
        body_args: body_args).send_message

```

### CODE REVIEW

File: app/models/state_file/automated_message/welcome.rb [hunk 1]
```diff
@@ -17,7 +17,8 @@ def email_subject(**args)
     end
 
     def email_body(**args)
+      binding.pry
```
Comment (rickreyhsig) [inline app/models/state_file/automated_message/welcome.rb:20]: REMOVE ME

File: app/models/state_file_az_intake.rb [hunk 2]
```diff
@@ -31,6 +31,7 @@
 #  household_excise_credit_claimed       :integer          default("unfilled"), not null
 #  last_sign_in_at                       :datetime
 #  last_sign_in_ip                       :inet
+#  locale                                :string           default("us")
```
Comment (tofarr) [inline app/models/state_file_az_intake.rb:34]: Should this default to `en`?

**Review by tofarr — APPROVED** [review]

File: app/models/state_file_az_intake.rb (hunk 2 above)
Comment (rickreyhsig) [reply app/models/state_file_az_intake.rb]: Done

File: app/services/state_file/messaging_service.rb [hunk 3]
```diff
@@ -1,9 +1,10 @@
 module StateFile
   class MessagingService
+    DEFAULT_LOCALE = 'en'
     attr_accessor :locale, :message, :intake, :submission, :locale, :message_tracker, :sent_messages
 
     def initialize(message:, intake:, submission: nil, locale: nil, sms: true, email: true, body_args: {})
-      @locale = locale || "en" # TODO: add intake.locale once we start collecting it
+      @locale = override_us_locale
```
[the remark this answers is not in the document]
Comment (rickreyhsig) [reply app/services/state_file/messaging_service.rb]: Done

**Comment by mpidcock** [comment]
@embarnard Did you finish reviewing this?

**Pull request merged**